mood-analysis.rb

```diff
@@ -14,11 +18,44 @@ def analyze_mood(words)
       sad += 1
     end
   end
+
   return ":-)" if happy > sad
   return ":-(" if happy < sad
   return ":-|"
 end
 
+def happy_days(result)
+  happy_days = 0
+  day_count = 0
+
+  result.each do |journal|
+
+    mood = journal[1]
+
+    if happy_days < 3
+      day_count += 1
+       happy_days += 1 if mood == ":-)"
+    end
+  end
+  day_count
+end
+
+
+def overall_mood(result)
+
+  overall_mood_result = {}
+  overall_mood_result[":-)"] = 0
+  overall_mood_result[":-|"] = 0
+  overall_mood_result[":-("] = 0
```

Comment:
If you do `overall_mood_result = Hash.new(0)` then you don't need to these three lines initing the key/values....